Essentials/src/com/earth2me/essentials/commands/Commandmute.java

```diff
@@ -40,20 +41,29 @@ public void run(final Server server, final CommandSource sender, final String co
                 throw new Exception(tl("muteExempt"));
             }
         }
+
+        long muteTimestamp = 0;
+        String time;
+
+        if (args.length > 1) {
+            time = args[1];
+            muteTimestamp = DateUtil.parseDateDiff(time, true);
```

Comment:
Nerp. Check out line 59 (old diff) - if this isn't a time, this will throw an exception. You don't catch it, so this will break the command.

Consider refactoring this so that this check happens in the same place as it did previously (line 56)

Reply:
My bad, this is an oversight on my part. I'll try to fix it soon. The reason for moving the code was so it can be checked before the mute event is fired. I think it makes sense that the mute event would not fire in this case if the timestamp is rejected.

Reply:
Oh actually I will note that I did test this PR before submitting it, and it worked (at least with valid time stamps)

Testing it now again, I'm actually still not able to break the command by typing in random gibberish. It gives the error "Error: Illegal date format."

I haven't looked into exactly how DateUtil works, but do you have an example of a command that would raise an exception?

Reply:
So that test is incorrect - if given random gibberish it should go into the mute reason, not error out.

Reply:
See for yourself: https://www.youtube.com/watch?v=OnYFKr3Twq8

Reply:
Is it still possible to set a mute reason without giving a duration if the command sender has permission to set unlimited mutes?

Reply:
Should be -- I'll be sure to check every combination when testing again. I'll admit I actually forgot about this issue for the last week, as I've been busy with other things. I will still be fixing the issue Joe mentioned. Let me know if you want it done sooner and I'll give it more priority, although I figure there's not a huge rush. :smiley:

Reply:
No worries - I haven't had a chance to look at PRs for a couple weeks but I finally got around to reviewing this one now.